src/thyra/ThyraAccount.ts

```diff
@@ -217,4 +229,138 @@ export class ThyraAccount implements IAccount {
   ): Promise<ITransactionDetails> {
     throw new Error('Method not implemented.');
   }
+
+  /**
+   * This method aims to interact with a smart contract deployed on the MASSA blockchain.
+   *
+   * @remarks
+   * If dryRun.dryRun is true, the method will dry run the smart contract call and return an
+   * IContractReadOperationResponse object which contains all the information about the dry run
+   * (state changes, gas used, etc.)
+   *
+   * @param contractAddress - The address of the smart contract.
+   * @param functionName - The name of the function to be called.
+   * @param parameter - The parameters as an Args object to be passed to the function.
+   * @param amount - The amount of MASSA coins to be sent to the block creator.
+   * @param dryRun - The dryRun object to be passed to the function.
+   *
+   * @returns if dryRun.dryRun is true, it returns an IContractReadOperationResponse object. Otherwise,
+   * it returns an ITransactionDetails object which contains the first event emitted by the contract.
+   * If the contract does not emit any event, it returns "Function called successfully but no event generated"
+   *
+   */
+  public async callSC(
```

Comment:
This function do a lot of things, can we split it with small functions. It would be easier to read

Reply:
is the new version ok for you ?